assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -86,23 +92,57 @@ const Piece = ({ piece, view }: { piece: Piece; view: "run" | "block" }) => (
   </>
 )
 
-const Trip = ({ trip }: { trip: Trip }) => {
-  const formattedVariant: string =
-    trip.viaVariant !== null && trip.viaVariant !== "_" ? trip.viaVariant : ""
-  const formattedRouteAndVariant: string =
-    trip.routeId !== null ? `${trip.routeId}_${formattedVariant}` : ""
-  return (
-    <Row
-      icon={questionMarkIcon()}
-      text={
-        <>
-          {formattedRouteAndVariant}{" "}
-          <span className="m-minischedule__headsign">{trip.headsign}</span>
-        </>
-      }
-      rightText={formattedScheduledTime(trip.startTime)}
-    />
-  )
+const Trip = ({
+  trip,
+  sequence,
+}: {
+  trip: Trip
+  sequence: "first" | "middle" | "last"
+}) => {
+  const startTime: string = formattedScheduledTime(trip.startTime)
+  if (isDeadhead(trip)) {
```

Comment:
@skyqrose this whole conditional is a little chunky for my taste, what if we extracted the deadhead case so this reads:

```
if (isDeadheadTrip) {
  return(deadheadRow(trip, sequence));
} else {
  // ...the stuff in the other branch
}
```